homeassistant/components/dhcp/__init__.py

```diff
@@ -271,3 +270,15 @@ def _decode_dhcp_option(dhcp_options, key):
 def _format_mac(mac_address):
     """Format a mac address for matching."""
     return format_mac(mac_address).replace(":", "")
+
+
+def _verify_l2socket_creation_permission():
+    """Create a socket using the scapy configured l2socket.
+
+    Try to create the socket
+    to see if we have permissions
+    since AsyncSniffer will do it another
+    thread so we will not be able to capture
+    any permission or bind errors.
+    """
+    conf.L2socket()
```

Comment:
Since conf.L2socket varies by system type I couldn't get it to patch so I wrapped it here.